python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/aaa-server-groups.j2

```diff
@@ -4,22 +4,20 @@
  that can be found in the LICENSE file.
 #}
 {# eos - aaa server groups #}
-{% if aaa_server_groups is arista.avd.defined %}
-{%     for aaa_server_group in aaa_server_groups %}
-{%         if aaa_server_group.type is arista.avd.defined and aaa_server_group.name is arista.avd.defined %}
+{% for aaa_server_group in aaa_server_groups | arista.avd.natural_sort('type') | arista.avd.natural_sort('name') %}
```

Comment:
Chained sorting works opposite of this. The output of each sort is just a list, so when you sort on `name` last, it will mix up all the types again. Maybe swapping the two will do what you need. Otherwise you can experiment with the `groupby` filter.

Reply:
Yes this is working opposite. I swap the sort vars. 

This throws error when any of the `type` and `name` is not defined. I think we need to make them required in schema.

Reply:
right. Previously we were just checking for type and name in the loop and ignoring entries without those two fields. I think it is a good idea to make the required in the schema and then remove the check inside the loop (since it will be enforced in natural_sort as well)

Reply:
@ClausHolbechArista Shall I create a new PR for change in schema?